aws/resource_aws_launch_configuration.go

```diff
@@ -355,8 +361,10 @@ func resourceAwsLaunchConfigurationCreate(d *schema.ResourceData, meta interface
 		vL := v.(*schema.Set).List()
 		for _, v := range vL {
 			bd := v.(map[string]interface{})
-			ebs := &autoscaling.Ebs{
-				DeleteOnTermination: aws.Bool(bd["delete_on_termination"].(bool)),
+			ebs := &autoscaling.Ebs{}
+
+			if v, ok := bd["no_device"].(bool); !ok && v {
+				ebs.DeleteOnTermination = aws.Bool(bd["delete_on_termination"].(bool))
```

Comment:
I'm not sure I follow what's happening here. Was this supposed to be a fix of existing code? i.e.

```go
if v, ok := bd["delete_on_termination"].(bool); ok {
    ebs.DeleteOnTermination = aws.Bool(v)
```
if not then can you please explain why can't we just leave the validation of potentially conflicting fields to the API?